packages/core/onClickOutside/index.ts

```diff
@@ -86,8 +86,18 @@ export function onClickOutside<T extends OnClickOutsideOptions>(
     handler(event)
   }
 
+  let isProcessingClick = false
+
   const cleanup = [
-    useEventListener(window, 'click', listener, { passive: true, capture }),
+    useEventListener(window, 'click', (event: PointerEvent) => {
+      if (!isProcessingClick) {
+        isProcessingClick = true
+        listener(event)
+        setTimeout(() => {
+          isProcessingClick = false
+        }, 0)
```

Comment:
```suggestion
        setTimeout(() => {
          isProcessingClick = false
        }, 0)
        listener(event)
```

Just in case listener throws an error, I think we should schedule the reset earlier